File: app/src/main/java/com/mcdev/whatsappunsavednumbers/MainActivity.java
package com.mcdev.whatsappunsavednumbers;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText inputNumberEt;
    Button startChatBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        init();



        startChatBtn.setOnClickListener(view -> {
            String phoneNumber = inputNumberEt.getText().toString();
            if (phoneNumber.isEmpty()) {
                Toast.makeText(this, "Phone number cannot be empty.", Toast.LENGTH_SHORT).show();
            } else {
                openWhatsApp(phoneNumber);
            }

        });

    }

    private void init() {
        inputNumberEt = findViewById(R.id.etInputNumber);
        startChatBtn = findViewById(R.id.btnStartChat);
    }

    private void openWhatsApp(String phoneNumber) {
        String url = "https://api.whatsapp.com/send?phone=" + phoneNumber;

        Intent openWhatsAppIntent = new Intent(Intent.ACTION_VIEW);
        openWhatsAppIntent.setData(Uri.parse(url));
        startActivity(openWhatsAppIntent);
    }
}

File: app/src/main/java/com/mcdev/whatsappunsavednumbers/MainActivity.kt
package com.mcdev.whatsappunsavednumbers

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.WindowManager
import android.widget.Toast
import android.content.Intent
import android.net.Uri
import android.util.Log
import android.view.LayoutInflater
import com.mcdev.whatsappunsavednumbers.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private val TAG = "MainActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val root = binding.root
        setContentView(root)
        //window.setFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS, WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)

        binding.btnStartChat.setOnClickListener {
            val countryCode = binding.countryCodePicker.selectedCountryCodeWithPlus
            Log.d(TAG, "onCreate: country code is $countryCode")
            val phoneNumber = countryCode + binding.etInputNumber.text.toString()
            if (phoneNumber.isEmpty() || phoneNumber.isBlank()) {
                Toast.makeText(this, R.string.phone_number_cannot_be_null, Toast.LENGTH_SHORT).show()
            } else {
                openWhatsApp(phoneNumber)
            }
        }
    }

    private fun openWhatsApp(phoneNumber: String) {
        Log.d(TAG, "onCreate: Starting chat with $phoneNumber")
        val url = getString(R.string.whatsapp_send_message_base_url) + phoneNumber
        val openWhatsAppIntent = Intent(Intent.ACTION_VIEW)
        openWhatsAppIntent.data = Uri.parse(url)
        startActivity(openWhatsAppIntent)
    }
}